platforms/core-runtime/launcher/src/main/java/org/gradle/launcher/daemon/client/DefaultDaemonStarter.java

```diff
@@ -141,7 +141,7 @@ public DaemonStartupInfo startDaemon(boolean singleUse) {
 
         LOGGER.debug("Using daemon args: {}", daemonArgs);
 
-        daemonArgs.add(GradleDaemon.class.getName());
+        daemonArgs.add("org.gradle.launcher.daemon.bootstrap.GradleDaemon");
```

Comment:
Use a string class name to remove the dependency on "daemon-main". It should be possible to remove this and use the executable Jar instead, will do this in a later PR.